pkg/dispatchcli/cmd/manage_context.go

```diff
@@ -21,48 +21,81 @@ import (
+	manageContextExample = i18n.T(`
+# list all contexts:
+$ dispatch manage context`)
 
-	// TODO: add examples
-	manageContextExample = i18n.T(``)
-	currentContext       = i18n.T(``)
+	currentContextLong = i18n.T(`Show current configuration context.`)
+
+	setContextLong    = i18n.T(`Set configuration context.`)
+	setContextExample = i18n.T(`
+# set current context to localhost:
+$ dispatch manage context set localhost
+Set context to localhost`)
+
+	deleteContextLong    = i18n.T(`Delete configuration context.`)
+	deleteContextExample = i18n.T(`
+# delete context localhost:
+$ dispatch manage context delete localhost
+Deleted context localhost`)
 )
 
 // NewCmdManageContext handles configuration context operations
 func NewCmdManageContext(out io.Writer, errOut io.Writer) *cobra.Command {
 	cmd := &cobra.Command{
-		Use:     "context [--set CONTEXT]",
+		Use:     "context",
 		Short:   i18n.T("Manage context"),
 		Long:    manageContextLong,
 		Example: manageContextExample,
-		Args:    cobra.ExactArgs(0),
 		Aliases: []string{"app"},
-		Run: func(cmd *cobra.Command, args []string) {
-			err := manageContext(out, errOut, cmd, args)
-			CheckErr(err)
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return formatContextOutput(out)
 		},
 	}
 
-	cmd.Flags().StringVarP(&currentContext, "set", "s", "", "Set current context")
+	cmd.AddCommand(NewCmdManageContextCurrent(out, errOut))
+	cmd.AddCommand(NewCmdManageContextSet(out, errOut))
+	cmd.AddCommand(NewCmdManageContextDelete(out, errOut))
+
 	return cmd
 }
 
-func manageContext(out, errOut io.Writer, cmd *cobra.Command, args []string) error {
-	if currentContext != "" {
-		if _, ok := cmdConfig.Contexts[currentContext]; !ok {
-			return errors.Errorf("No such context %s", currentContext)
-		}
-		cmdConfig.Current = currentContext
-		b, _ := json.MarshalIndent(cmdConfig, "", "    ")
-		path := viper.ConfigFileUsed()
-		err := ioutil.WriteFile(path, b, 0644)
-		if err != nil {
-			return errors.Errorf("Failed to write config file %s", path)
-		}
-		fmt.Fprintf(out, "Set context to %s\n", currentContext)
-		return nil
+func setContext(out, errOut io.Writer, cmd *cobra.Command, args []string) error {
+	currentContext := args[0]
+	if _, ok := cmdConfig.Contexts[currentContext]; !ok {
+		return errors.Errorf("No such context %s", currentContext)
 	}
-	return formatContextOutput(out)
+	cmdConfig.Current = currentContext
+	b, _ := json.MarshalIndent(cmdConfig, "", "    ")
+	path := viper.ConfigFileUsed()
+	err := ioutil.WriteFile(path, b, 0644)
+	if err != nil {
+		return errors.Errorf("Failed to write config file %s", path)
+	}
+	fmt.Fprintf(out, "Set context to %s\n", currentContext)
+	return nil
+}
+
+func deleteContext(out, errOut io.Writer, cmd *cobra.Command, args []string) error {
+	ctxName := args[0]
+	if _, ok := cmdConfig.Contexts[ctxName]; !ok {
+		return errors.Errorf("No such context %s", ctxName)
+	}
+
+	if cmdConfig.Current == ctxName {
```

Comment:
What if there is no other context?

Anyway, I would prefer to delete the current active context. We can then display an error message saying no context has been set every time user executes a command. Also, ensure the error is only thrown when the config file is present and the context is nil because we do have usecase like install/single-binary which you can run without a config.json file.

Reply:
Delete active context is enabled. For now, viper will try to load config file, if current context was deleted from contexts, cli will try to read values from flags. The 'current context was found' msg will be displayed using 'context current' cmd. So other use cases should not be affected.